### ISSUE TITLE
Add TypeScript type definition

### ISSUE THREAD

**Issue by greypants** [ten1seven/what-input#106]
Hey @ten1seven  - we're converting to typescript, and have typed what-input locally. I'll open a PR soon to add this to your package so it ships fully typed!

This is what we have locally - i need to double check the API and make sure it's correct and complete. PR forthcoming...

```ts
declare module "what-input" {
  export interface WhatInput {
    ask(intent?: string): string | null;
    ignoreKeys(keys: (string | number)[]): void;
    registerOnChange(callback: () => void, intent: string): void;
    unRegisterOnChange(callback: () => void): void;
  }

  const whatInput: WhatInput;

  export default whatInput;
}
```

**Comment by ten1seven** [comment]
Oh, cool! I'm excited!

**Comment by greypants** [comment]
Resolved by #107
